### PR TITLE
LPD-2612 | master

### PR DESCRIPTION
**Pull request by marcosapmf**
Jira ticket: https://liferay.atlassian.net/browse/LPD-2612

Hello Core Infra Team.

I'm sending this pull request to you because of this [comment](https://liferay.atlassian.net/browse/LPD-2612?focusedCommentId=2669143). 
@victorhcunha  had [reverted](https://github.com/brianchandotcom/liferay-portal/pull/151087) this changes due a performance regression issue. 
In light of this, I'm re sending this pull request with some improvements and I'd be glad if you guys could review it.

Summarizing, the main changes are:

1. Decreasing the priority of checking specific resource permissions. I'm only checking that after `analyticsReportsInfoItem.isShow(analyticsReportsInfoItemObject)` returns `false`
2. Only check the resource permissions if the user is logged in, otherwise return `false`
3. Prevent checking a resource permission more than once when many instances of the same portlet are in a page
4. Adding a cache layer for resource permission checker

Please let me know if you have questions about this :smile: 

Thank you

### CODE REVIEW

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java [hunk 1]
```diff
@@ -245,6 +250,14 @@ public boolean isShow(HttpServletRequest httpServletRequest)
 	protected void activate() {
 		_portletNamespace = _portal.getPortletNamespace(
 			AnalyticsReportsPortletKeys.ANALYTICS_REPORTS);
+
+		CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+
+		cacheBuilder.expireAfterAccess(10, TimeUnit.MINUTES);
```
Comment (dantewang) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java:256]: @marcosapmf Currently the values in this cache will only be invalided due to this 10-minute expiration. So, if the permission gets changed, the change will only take effect after this cache proactively expire the previous result. If there's no proper way to invalidate cache values upon permission changes, please do not cache it.

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java [hunk 2]
```diff
@@ -5,6 +5,9 @@
 
 package com.liferay.analytics.reports.web.internal.product.navigation.control.menu;
 
+import com.google.common.cache.Cache;
```
Comment (dantewang) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/AnalyticsReportsProductNavigationControlMenuEntry.java:8]: @marcosapmf Also, please avoid using Google Cache even if you can find a way to properly manage the result invalidation. Please use portal cache instead. With portal cache (based on Ehcache) we can easily review the hit/miss/other statistics, while with Google Cache we don't have any of those.

**Review by dantewang — CHANGES_REQUESTED** [review]
@marcosapmf Please see my comments and resend, thanks!

**Comment by marcosapmf** [comment]
Hi @dantewang,
thanks for reviewing.
I just resent another pull request removing cache layer: https://github.com/liferay-core-infra/liferay-portal/pull/7122
Thanks

**Pull request closed without merging**